### PR TITLE
github: Replaced run-action with direct call of pre-commit in lint CI

### PR DESCRIPTION
**Pull request by gemesa**
## Detailed description

There are some warnings going on in the action runs, see https://github.com/blackmagic-debug/blackmagic/actions/runs/4802501164 for example:

```
Annotations
4 warnings
pre-commit
Node.js 12 actions are deprecated. Please update the following actions to use Node.js 16: before-commit/run-action@v2.0.3. For more information see: https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/.
pre-commit
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
pre-commit
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
pre-commit
The `save-state` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
```

This can not be fixed with a version bump because there are no new releases/tags being made since Apr 21, 2021 in [before-commit/run-action](https://github.com/before-commit/run-action/tags). We can install and call `pre-commit` directly though instead as we see what options are used in [this run](https://github.com/blackmagic-debug/blackmagic/actions/runs/4802501164/jobs/8546036072):

```
/home/runner/.local/bin/pre-commit run --show-diff-on-failure --color=always --all-files
```

Compare these 2 runs and you can see they are identical:

before (this repo): https://github.com/blackmagic-debug/blackmagic/actions/runs/4802501164/jobs/8546036072
after (my forked repo): https://github.com/gemesa/blackmagic/actions/runs/4809530379/jobs/8560906932

## Your checklist for this pull request

* [x] I've read the [Code of Conduct](https://github.com/blackmagic-debug/blackmagic/blob/main/CODE_OF_CONDUCT.md)
* [x] I've read the [guidelines for contributing](https://github.com/blackmagic-debug/blackmagic/blob/main/CONTRIBUTING.md) to this repository
* [x] It builds for hardware native (`make PROBE_HOST=native`)
* [x] It builds as BMDA (`make PROBE_HOST=hosted`)
* [x] I've tested it to the best of my ability
* [x] My commit messages provide a useful short description of what the commits do

### CODE REVIEW

**Review by dragonmux — APPROVED** [review]
This looks reasonable enough to us.. unless @perigoso (who originally contributed this) knows of any reason we shouldn't merge this, we will

**Comment by perigoso** [comment]
The reasons I went with the before-commit fork of pre-commit originally were, let's say, not of technical nature (@dragonmux should be aware but if not we can talk about it), that said, the author and maintainer of before-commit didn't have time to fully commit to the project and it's mostly unmaintained now.
These deprecations could be easily fixed, there's not a lot going with the action even upstream, but we might just go ahead and use pre-commit and save the headaches...

That said, this is not what this PR is doing, this is installing an running the command directly, we could do that too but I would prefer using the pre-commit action (assuming they fixed the deprecations, I will have to check)

**Comment by dragonmux** [comment]
Given Perigoso's PR uses a GH Action instead of manually doing the installation steps, which means less for us to maintain that can go wrong, we're going to take his PR instead of this. We're very pleased you bought this to our attention and came up with a working fix all the same - thank you for the contribution!

**Pull request closed without merging**
